app/mailers/application_mailer.rb

```diff
@@ -6,11 +6,14 @@ class ApplicationMailer < ActionMailer::Base
           :from => proc { Setting[:email_reply_address] || "noreply@foreman.example.org" }
   after_action :set_delivery_options
 
+  helper :mailer
+
   def mail(headers = {}, &block)
     if headers.present?
       headers[:to] = utf8_encode(headers[:to])
       headers[:subject] = "#{Setting[:email_subject_prefix]} #{headers[:subject]}" if (headers[:subject] && Setting[:email_subject_prefix].present?)
       headers['X-Foreman-Server'] = URI.parse(Setting[:foreman_url]).host if Setting[:foreman_url].present?
+      headers['X-Foreman-ID'] = Setting[:instance_title] if Setting[:instance_title].present?
```

Comment:
I think title is more human readable and would expect the UUID in the ID. Perhaps even explicit
```suggestion
      headers['X-Foreman-ID'] = Setting[:instance_id] if Setting[:instance_id].present?
```